api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -64,74 +55,50 @@ public boolean deleteInstitution(final String shortName) {
   public Optional<Institution> updateInstitution(
       final String shortName, final Institution institutionToUpdate) {
     return getDbInstitution(shortName)
-        .map(dbInst -> toModelClass(saveInstitution(institutionToUpdate, dbInst)));
+        .map(dbInst -> toModel(institutionDao.save(updateDbObject(dbInst, institutionToUpdate))));
   }
 
   private Optional<DbInstitution> getDbInstitution(String shortName) {
     return institutionDao.findOneByShortName(shortName);
   }
 
-  private DbInstitution saveInstitution(final Institution modelClass, final DbInstitution dbClass) {
-    dbClass.setShortName(modelClass.getShortName());
-    dbClass.setDisplayName(modelClass.getDisplayName());
-    dbClass.setOrganizationTypeEnum(
-        DbStorageEnums.organizationTypeToStorage(modelClass.getOrganizationTypeEnum()));
-    dbClass.setOrganizationTypeOtherText(modelClass.getOrganizationTypeOtherText());
-
-    return saveInstitutionEmailPatterns(modelClass, institutionDao.save(dbClass));
+  private DbInstitution newDbObject(final Institution modelObject) {
```

Comment:
Model <-> API conversions are now fully independent of DB state!

TODO (not in this PR) move conversions out of here into a MapStruct mapper.